### ISSUE TITLE
Updated / improved bazel examples & documentation

### ISSUE THREAD

**Issue by drewolson** [google/dagger#1076]
I recently created [this repository](https://github.com/drewolson/dagger_bazel_example) as an exercise to build a minimal working example with dagger 2 and bazel. There were a few small differences from the documentation that took a lot of time to resolve.

1. It was necessary to use [strip_prefix](https://github.com/drewolson/dagger_bazel_example/blob/dbd50982653a97b9bba4e14454ccc215fd0acf3d/WORKSPACE#L64) in the `http_archive` task in order to include the dependency as stated in the documentation. Not doing so resulted in [this error](https://gist.github.com/drewolson/93d8bf9c5d7eb6b4b02cc2f3cadf5c5e#file-output-txt).

2. It was necessary to use the [shaded version of errorprone javac](https://github.com/drewolson/dagger_bazel_example/blob/dbd50982653a97b9bba4e14454ccc215fd0acf3d/WORKSPACE#L53). Not doing so resulted in a [this error](https://gist.github.com/drewolson/93d8bf9c5d7eb6b4b02cc2f3cadf5c5e#file-new_output-txt).

I think it would be useful to update the documentation to include the `strip_prefix` argument in the `http_archive` snippet and also provide a full example using bazel. Thoughts?

**Comment by drewolson** [comment]
Any interest in making these documentation improvements? If so, I'd be happy to PR something. Otherwise I'll go ahead and close this issue.

**Comment by ronshapiro** [comment]
Re: `strip_prefix` - I intentionally left that out, since some people may want to use a different type of workspace rule, and I didn't want them to be confused that that was necessary. Maybe that's not the case though, and it's more confusing without it.

Regarding the shaded-javac, that's definitely surprising, since we are including that. Actually, it looks like your workspace file has a newer google-java-format, which might mean that you needed a newer shaded-javac jar?

**Comment by drewolson** [comment]
@ronshapiro re: `strip_prefix`, I think it's worth making sure that whatever snippet is included works without modification. I think it would be worthwhile to include `strip_prefix` and then note that other types of workspace rules are possible as well. At least that way, folks who just copy the example will end up with working code.

I'm not sure why `shaded-javac` was required for me. Even after using the `http_archive` rule, I was forced to require each dependency one-by-one (standard for bazel). In each case, I simply picked the newest available version on maven central that was compatible. Because bazel requires this explicitly listing of third-party dependencies, I figured a full, working example with a concrete list could be helpful to users.
